File: BACK/API.Library/Infrastructure/Library.Infrastructure/Entities/Item.cs
using Library.Infrastructure.Entities.Base;
using Library.Infrastructure.Entities.Enumerations;

namespace Library.Infrastructure.Entities;

/// <summary>
///     An existing item that a user can add to its collections.
/// </summary>
public class Item : Entity
{
    public Item(string label, DateTime releaseDate, string? support, string? imageURL)
    {
        Label = label;
        ReleaseDate = releaseDate;
        Support = support;
        ImageURL = imageURL;
    }

    /// <summary>
    ///     Label of the item.
    /// </summary>
    public string Label { get; set; }

    /// <summary>
    ///     Type of item.
    /// </summary>
    public ItemType Type { get; set; }

    /// <summary>
    ///     Release date of the item.
    /// </summary>
    public DateTime ReleaseDate { get; set; }

    /// <summary>
    ///     Support of the item (physical, digital, etc...)
    /// </summary>
    public string? Support { get; set; }

    /// <summary>
    ///     URL to an image of the item.
    /// </summary>
    public string? ImageURL { get; set; }

    public List<UserItem> UserItems { get; set; } = default!;
}

File: FRONT/APP.Mobile/vaultapp/lib/app/modules/items/data/models/item.dart
class Item {
  Item(
      {this.id,
      required this.label,
      required this.type,
      required this.releaseDate,
      this.support,
      this.imageURL});

  String? id;
  final String label;
  final String type;
  final DateTime releaseDate;
  final String? support;
  final String? imageURL;

  factory Item.fromJson(Map<String, dynamic> json) {
    String? id = json['id'];
    String label = json['label'];
    String type = json['type'];
    String? support = json['support'];
    String? imageURL = json['imageURL'];

    DateTime releaseDate = DateTime.parse(json['releaseDate']);

    return Item(
        id: id,
        label: label,
        type: type,
        releaseDate: releaseDate,
        support: support,
        imageURL: imageURL);
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'label': label,
      'type': type,
      'releaseDate': releaseDate.toIso8601String(),
      'support': support,
      'imageURL': imageURL
    };
  }
}
